src/alitra/frame_dataclasses.py

```diff
@@ -9,32 +9,37 @@
 
 @dataclass
 class Transform:
-    """A transform object that describe the transformation between two frames.
-    Translations must be expressed in the (to_) frame"""
+    """
+    A transform object that describe the transformation between two frames.
+    Euler or quaternion must be provided to perform a rotation. If no rotation is required specify the unit quaternion
+    or zero Euler angles. Translations must be expressed in the (to_) frame
+    """
 
     translation: Translation
-    euler: Euler
     from_: Literal["robot", "asset"]
     to_: Literal["asset", "robot"]
-    rotation_object: Rotation = Rotation.from_euler("zyx", [0, 0, 0])
+    rotation_object: Rotation = None
+    euler: Euler = None
+    quaternion: Quaternion = None
 
     def __post_init__(self):
-        if not self.translation.from_ == self.euler.from_ == self.from_:
+        if (
+            not self.translation.from_ == self.from_
+            and not self.translation.to_ == self.to_
+            or not self.to_ == self.translation.frame
+        ):
```

Comment:
Removed the checks on which from/to frames are set in the Euler dataclass. 

To me it made more sense to only have a from/to association within the `Transform` object. As that is a transform from some frame to some other frame. The quaternion itself I though of as having a frame, not two frames. 

I guess there is a distinction on what the quaternion/euler angle is used for. If it's used as the basis for a rotation it might make sense to have a from/to. While if it just represents an orientation it would make sense to have a frame.

Reply:
My point and suggestion then: To remove from/to in the Euler class and have from/to only in the Transform class. Then the Quaternion and Euler will only contain a frame.

This might seem weird when specifying a frame transform based on a quaternion/euler, as it doesn't necessarily have one fram then. Thoughts?

Reply:
It could be accommodated if we made a new data-class that is the angular version of "translation". 
e.g., "rotation".

But I think we might remove it as you suggest, and assume that it is used correctly.